h5netcdf/tests/test_h5netcdf.py

```diff
@@ -482,9 +481,39 @@ def test_reading_str_array_from_netCDF4(tmp_netcdf):
     assert array_equal(v, _string_array)
     ds.close()
 
-
 def test_nc_properties(tmp_netcdf):
     with h5netcdf.File(tmp_netcdf, 'w') as ds:
         pass
     with h5py.File(tmp_netcdf, 'r') as f:
         assert 'h5netcdf' in f.attrs['_NCProperties']
+
+
+def _silent_remove(tmp_netcdf):
+    # http://stackoverflow.com/a/10840586
+    import os
+    try:
+        os.remove(tmp_netcdf)
+    except OSError:
+        pass
+
+def test_failed_read_open_and_clean_delete(tmp_netcdf):
+    # A file that does not exist but is opened for
+    # reading should only raise an IOError and
+    # no AttributeError at garbage collection.
+    try:
+        _silent_remove(tmp_netcdf)
+        with h5netcdf.File(tmp_netcdf, 'r') as ds:
+            pass
+    except IOError:
+        pass
+
+    # Look at garbage collection:
+    import gc
```

Comment:
Done! About the extra commits, I know. Still getting used to the complete workflow. A few more pull requests and it should be seamless :)